homeassistant/components/generic/camera.py

```diff
@@ -132,6 +133,11 @@ async def async_camera_image(
         self, width: int | None = None, height: int | None = None
     ) -> bytes | None:
         """Return a still image response from the camera."""
+        if not self._still_image_url:
+            if self.stream:
+                return await self.stream.get_image(width=width, height=height)
+            await self.create_stream()
```

Comment:
@allenporter This will currently attempt to create a new stream if one is not present, but will return None afterwards. Two questions:
1) Should we try and make it return an image by waiting for an image to be created before returning? I'm thinking it's not worth the hassle.
2) Do we need to put a lock in Camera.create_stream()? It doesn't look like there is anything there to prevent multiple streams from being created concurrently.

Reply:
Agreed that if we're starting a stream, it may take a bit long to get an image from it.  Though that is effectively what the ffmpeg call does now.  Would another option be to only use if an existing stream is already present? (E.g. it was preloaded already)

Using a lock in the existing call makes sense to me. I think the alternative is to re-use an existing awaitable if one already exists -- and a lock sounds simpler.   Have you thought about how timeouts are handled?

Reply:
Yes, we could use it only if an existing stream is present. But that brings up your earlier question in the other PR - which component is responsible for managing the stream? I'm thinking it should be the specific camera which is in this case generic. If someone was to use generic without `preload_stream`, doing it the current way would still allow them to get an image on the second request to the image (assuming the stream starts up in time).
Which timeouts are you looking at?

Reply:
Ah timeouts are a non-issue, i was confused, thinking of the case where it was waiting for a response from the worker which does not happen.

Can `camera` manage this for all cameras as a default implementation?

Reply:
For example in `nest` [this call](https://github.com/home-assistant/core/blob/6e13605cada2dea7aa55d57f55ee50546a23727f/homeassistant/components/nest/camera_sdm.py#L228) could be replaced with `super().async_camera_image`

Reply:
Maybe nest would change to stop using the event, and only use the still image from the stream, given i added better support for getting snapshots from events recently using a different API.  Maybe the logic is flipped in nest to work the way generic camera works: if the parent has an implementation from stream, use it, otherwise use the still image url as a fallback?

Reply:
Yes, it's probably better to do this in `camera`. The individual camera implementations will still have to make changes if they wish to incorporate this functionality.

Reply:
I don't quite follow nest vs generic. I assume generic should try still_image_url first (that's what's done in this PR).

Reply:
Oops, i missed the still image check at the top.

Reply:
Had a look at moving this into into `camera`. The problem there is that cameras have that dual sync/async setup where `Camera.camera_image` is not implemented and `Camera.async_camera_image` schedules a thread to call it via `async_add_executor_job`. Then in `generic` itself, `Generic.async_camera_image` is implemented while the sync `Generic.camera_image` is a wrapper that calls it from the main loop with `run_coroutine_threadsafe`. There is a thread->asyncio->thread usage that is warned against here: https://developers.home-assistant.io/docs/asyncio_working_with_async/#calling-async-functions-from-threads which we have to guard against regardless, but this gets more complicated if we move the functionality up to `super` as there's another layer of sync<->async kluge.

Reply:
Do you think there is an easy workaround or is it time to bite the bullet and do all the work to get rid of the sync / async bridges?

Reply:
I'm not sure why the API has to have sync and async functions. Legacy? I don't think we can just remove it without an architecture discussion. Even if we clean up the stuff in the core we might break some custom components.
I'll push some changes here which illustrate the problem more clearly.

Reply:
Yeah, it looks like legacy to me.  Yeah, i think it needs a deprecation log warning saying it will be removed in a few versions -- so doing the cleanup is fine, but maybe doesn't help today.